R/zchunk_L232.industry.R

```diff
@@ -89,295 +85,554 @@ module_energy_L232.industry_DISABLED <- function(command, ...) {
+    # Instead, interpolate the coefficients to these global default values in a specified period
+    # in below pipeline, year 2150 refers to `indcoef_conv_year` in the old chunk.
+    A32.globaltech_coef %>%
+      repeat_add_columns(tibble(GCAM_region_ID = GCAM_region_names[["GCAM_region_ID"]])) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(select(filter(L232.StubTechCoef_industry_base, year == max(BASE_YEARS)), region, minicam.energy.input, coefficient),
+                               by = c("region", "minicam.energy.input")) %>%
+      rename(`2010` = coefficient) %>%
+      mutate(`2150` = `2100`) %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(supplysector, subsector, technology, minicam.energy.input, GCAM_region_ID, region),
+               year = c(max(BASE_YEARS), FUTURE_YEARS, 2150)) %>%
+      arrange(supplysector, subsector, technology, minicam.energy.input, GCAM_region_ID, region, year) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input, GCAM_region_ID, region) %>%
+      mutate(coefficient = approx_fun(year, value),
+             coefficient = round(coefficient, energy.DIGITS_COEFFICIENT)) %>%
+      ungroup %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      mutate(stub.technology = technology,
+             market.name = region) %>%
+      # Combine the base year and future year coefficients
+      select(one_of(LEVEL2_DATA_NAMES[["StubTechCoef"]])) %>%
+      bind_rows(select(L232.StubTechCoef_industry_base, one_of(LEVEL2_DATA_NAMES[["StubTechCoef"]]))) %>%
+      mutate(coefficient = round(coefficient, energy.DIGITS_COEFFICIENT)) ->
+      L232.StubTechCoef_industry
+
+    # L232.FuelPrefElast_indenergy: fuel preference elasticities of industrial energy use
+    # First, calculate the fuel shares allocated to each fuel
+    L232.in_EJ_R_indenergy_F_Yh %>%
+      group_by(region, GCAM_region_ID, supplysector, subsector, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup ->
+      L232.indenergy_fuel_shares # intermediate tibble
+
+    L232.indenergy_fuel_shares %>%
+      group_by(region, GCAM_region_ID, supplysector, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      right_join(L232.indenergy_fuel_shares, by = c("region", "GCAM_region_ID", "supplysector", "year")) %>%
+      rename(total = value.x,
+             value = value.y) %>%
+      mutate(share = value/total) %>%
+      filter(year == max(BASE_YEARS)) %>%
+      mutate(fuelprefElasticity = 0) ->
+      L232.indenergy_fuel_shares
+
+    # Set fuel preference elasticities as indicated by exogenous rules
+    for( j in 1:nrow( L232.indenergy_fuel_shares ) ){
+      for( i in 1:nrow( A32.fuelprefElasticity ) ){
+        L232.indenergy_fuel_shares$fuelprefElasticity[j][
```

Comment:
Kind of confusing. I'm sure there's a more efficient way to do this but don't have time to drill down into this right now. @fengly20 please remind me to come back to this before we merge this PR.

Reply:
It took me a while to figure out what this code was doing, though after going through it seems pretty simple. The description line you added in the A32.fuelprefelasticity.csv file helps, but I would like it if you added a sentence or two in this code explaining what it does rather than saying"by exogenous rule"